core/src/main/java/org/apache/iceberg/UpdateRequirement.java

```diff
@@ -199,4 +220,56 @@ public void validate(TableMetadata base) {
       }
     }
   }
+
+  class AssertLastAssignedViewVersionId implements UpdateRequirement {
```

Comment:
What is the purpose of this validation?

For tables, we have situations where we need to assign a new ID, like `AddSchema`. In those cases, we let the service assign the ID instead of sending one so that we don't need to assert that the previous set of IDs had some maximum that we incremented. Instead, the service assigns the ID and the `SetCurrentSchema` change is sent with a `-1` to use the last assigned ID.

Reply:
it was mainly to express monitonically increasing view version IDs, but this can also be expressed via `AssertCurrentViewVersionID`. However, the error reporting with `AssertCurrentViewVersionID` isn't very good in indicating that view version IDs aren't monotonically increasing

Reply:
Why do these need to be monotonically increasing? I don't think we care about how view version IDs are assigned, as long as there aren't duplicates.